File: iks-core/src/com/iks/cms/core/sql/Column.java
package com.iks.cms.core.sql;

/**
 * @author Igor Kaynov
 */
public class Column {
  private String     tableAlias;
  private String     name;
  private String     alias;
  private ColumnType type = ColumnType.String;
  public Column( String name ) {
    this.name = name;
  }
  public Column( Table table, String name ) {
    this( table.getAlias(), name, null );
  }
  public Column( Table table, String name, String alias ) {
    this( table.getAlias(), name, alias );
  }
  public Column( String tableAlias, String name, String alias ) {
    this( name );
    this.tableAlias = tableAlias;
    this.alias = alias;
  }
  public String getName() {
    return name;
  }
  public String toString() {
    StringBuilder sb = new StringBuilder();
    if( tableAlias != null ) {
      sb.append( tableAlias ).append( '.' );
    }
    sb.append( name );
    if( alias != null ) {
      sb.append( " as " ).append( alias );
    }
    return sb.toString();
  }
  public ColumnType getType() {
    return type;
  }
  public void setType( ColumnType type ) {
    this.type = type;
  }
}

File: iks-core/src/com/iks/cms/core/sql/Column.kt
package com.iks.cms.core.sql

import java.util.*

/**
 * @author Igor Kaynov
 */
class Column(val name: String) {
  var tableAlias: String? = null
  var alias: String? = null
  var type = ColumnType.String

  constructor(table: Table, name: String) : this(table.alias, name, null) {
  }

  constructor(table: Table, name: String, alias: String?) : this(table.alias, name, alias) {
  }

  constructor(tableAlias: String?, name: String, alias: String?) : this(name) {
    this.tableAlias = tableAlias
    this.alias = alias
  }

  override fun toString(): String {
    val sb = StringBuilder()
    if (tableAlias != null) {
      sb.append(tableAlias.quote()).append('.')
    }
    sb.append(name.quote())
    //    if( alias != null ) {
    //      sb.append( " as " ).append( alias );
    //    }
    return sb.toString()
  }

  override fun equals(other: Any?): Boolean {
    if (other == null || other !is Column) {
      return false
    }
    return tableAlias == other.tableAlias && name == other.name
  }
}
